pkg/network/server.go

```diff
@@ -80,21 +82,27 @@ func randomID() uint32 {
 }
 
 // NewServer returns a new Server, initialized with the given configuration.
-func NewServer(config ServerConfig, chain core.Blockchainer) *Server {
+func NewServer(config ServerConfig, chain core.Blockchainer, log *zap.Logger) *Server {
+	if log == nil {
+		log = zap.NewNop()
```

Comment:
I think we're better being consistent about all of the `new*` functions, at the moment `NewBlockchain` fails with `nil` logger and all the others succeed. IMO, it's OK to succeed, but this behavior should be the same for all functions like this.